From 6346971db1bc310e58c086d28e472c91e9c4e920 Mon Sep 17 00:00:00 2001
From: Jon TURNEY <jon.turney@dronecode.org.uk>
Date: Fri, 19 Jun 2009 21:14:47 +0100
Subject: [PATCH 17/44] Copy the state of the Windows keyboard device to the Virtual Core Keyboard at startup.

Otherwise, this happens lazily after the first keypress, which can lead 
to applications which are started from a shell window and inspect the 
keyboard state before a character is typed getting the wrong idea about 
the desired keymap (e.g. xemacs shows this behaviour)

Signed-off-by: Jon TURNEY <jon.turney@dronecode.org.uk>
---
 hw/xwin/winkeybd.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/hw/xwin/winkeybd.c b/hw/xwin/winkeybd.c
index 611ea5d..357d4ad 100644
--- a/hw/xwin/winkeybd.c
+++ b/hw/xwin/winkeybd.c
@@ -252,6 +252,10 @@ winKeybdProc (DeviceIntPtr pDeviceInt, int iState)
       
     case DEVICE_ON: 
       pDevice->on = TRUE;
+
+      // immediately copy the state of this keyboard device to the VCK
+      // (which otherwise happens lazily after the first keypress)
+      CopyKeyClass(pDeviceInt, inputInfo.keyboard);
       break;
 
     case DEVICE_CLOSE:
-- 
1.6.4.2

